src/components/Navbar/utils.ts

```diff
@@ -10,3 +10,52 @@ export const getNavbarPagesUrls = () => {
     [NavbarPages.GOVERNANCE]: config.get('GOVERNANCE_URL')
   }
 }
+
+export type NavbarExtraButton = {
+  text: string
+  link: string
+  visible: boolean
+  textColor?: `#${string}`
+  backgroundColor?: `#${string}`
+  id?: string
+  ttl: number
+}
+
+export type LocalStorageNavbarExtraButton = {
+  button: NavbarExtraButton
+  expiresAt: number
+}
+
+export const getExtraButton = async () => {
+  const cachedExtraButton = localStorage.getItem('navbarExtraButton')
+  if (cachedExtraButton) {
+    try {
+      const parsed = JSON.parse(
+        cachedExtraButton
+      ) as LocalStorageNavbarExtraButton
+      if (parsed.expiresAt > Date.now()) {
+        return parsed.button
+      }
+    } catch (error) {
+      // error parsing cached data, ignore and fetch from Contentful
+    }
+  }
```

Comment:
This above restores the data from a localStorage cache, to avoid hitting contentful every time. The TTL is configured form contentful itself.